solutions/instances/main.tf

```diff
@@ -244,7 +239,7 @@ module "cos_bucket" {
   providers = {
     ibm = ibm.cos
   }
-  count   = (length(coalesce(local.bucket_config_map, [])) != 0) ? 1 : 0 # no need to call COS module if consumer is using existing COS bucket
+  count   = (var.log_analysis_enable_archive || var.enable_at_event_routing_to_cos_bucket) && (length(coalesce(local.bucket_config_map, [])) != 0) ? 1 : 0 # no need to call COS module if consumer is using existing COS bucket
```

Comment:
do we need to add `var.log_analysis_enable_archive || var.enable_at_event_routing_to_cos_bucket` in the condition still ....after making code changes for bucket_config_map at line 46 ?